src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -1356,6 +1364,7 @@ public void Dispose()
         }
 
         private Dictionary<Type, (Action<Array> Releaser, List<Array> Array)> _pooledArrays = null;
+        private IBlittableDocumentModifier _modifier;
```

Comment:
Shouldn't this be a list? What if someone would call `context.CreateBlittableMetadataModifier` twice on the same `context`? Then we won't clear allocations of the first one.

Reply:
@arekpalinski there are plenty of places in the code where this happens, let me explain the reasoning for not clearing it.
At first i thought i should throw if the merger is not null or dispose of the old one but context might be reset in two cases once when it is returned to the pull and second when somebody is manualy invoking reset (and in this case it might not be correct to replace the merger) i left the responsibility to whom ever writes the code. 
there are only 3 places in our code using a modifier there was a lot of consideration not to shake things up to strongly.